common/v2/features/Dashboard/components/TokenPanel/TokenPanel.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState, useContext, useEffect } from 'react';
+
+import { StoreContext, RatesContext, TokenInfoService } from 'v2/services';
+import { TTicker, AssetWithDetails, ExtendedAsset } from 'v2/types';
+import { TokenList } from './TokenList';
+import { TokenDetails } from './TokenDetails';
+import { AddToken } from './AddToken';
+
+export function TokenPanel() {
+  const [showDetailsView, setShowDetailsView] = useState(false);
+  const [showAddToken, setShowAddToken] = useState(false);
+  const [currentToken, setCurrentToken] = useState<AssetWithDetails>();
+  const [allTokens, setAllTokens] = useState<AssetWithDetails[]>([]);
+  const [isScanning, setIsScanning] = useState(false);
+
+  const { accounts, totals, currentAccounts, scanTokens } = useContext(StoreContext);
+  const { getRate, rates } = useContext(RatesContext);
+
+  const handleScanTokens = async (asset?: ExtendedAsset) => {
+    try {
+      setIsScanning(true);
+      await scanTokens(asset);
+      setIsScanning(false);
+    } catch (e) {
+      setIsScanning(false);
+    }
+  };
+
+  useEffect(() => {
+    const getTokensWithDetails = async () => {
+      const selectedAccounts = currentAccounts();
+      const tokenTotals = totals(selectedAccounts);
+
+      // Fetch details for all tokens of selected accounts
+      const tokensDetails: any[] = await TokenInfoService.instance.getTokensInfo(
+        tokenTotals.map(x => x.contractAddress!).filter(x => x)
+      );
+
+      // Add token details and token rate info to all assets that have  a contractAddress
+      const tempTokens = totals(selectedAccounts).reduce((tokens: AssetWithDetails[], asset) => {
+        if (asset.contractAddress) {
+          const existingTokenDetails = tokensDetails.find(
+            details => details.address === asset.contractAddress
+          );
+
+          tokens.push(
+            Object.assign(asset, {
+              rate: getRate(asset.ticker as TTicker) || 0,
+              details: existingTokenDetails || {}
+            })
+          );
+        }
+        return tokens;
+      }, []);
```

Comment:
I think we should retrigger the effect at least when `currrentAccounts` changes.

Reply:
It is retriggered if accounts or rates is changed (https://github.com/MyCryptoHQ/MyCrypto/pull/2742/commits/ea3272bb8cbee1e945c8eaad0cdf348d1fb88b95#diff-29388158806ca87065463442d4ee4c56L60), but this empty array here is initial value for reduce.

Reply:
mea culpa!